### PR TITLE
zeek: fix handling of zeek.http.host when URI contains a domain

### LINKED ISSUE #6913: logs-zeek.http-2.11.0 uri-parts
logs-zeek.http pipeline seems to error on rename zeek.http.host to url.domain as it already contains a value derived from the uri-parts processor. the remove "url.domain" is set to only run when its empty so if already filled it will error.

Why is it there as url-parts can generate it?

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

It seems that some zeek documents will retain the domain part in the URI field, causing an ingest pipeline failure when renaming `zeek.http.host` to `url.domain`. So conditionally rename `zeek.http.host` to `url.domain` only if `url.domain` is not present. Retain `zeek.http.host` if differs from `url.domain`.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Fixes #6913

### CODE REVIEW

File: packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log [hunk 1]
```diff
@@ -7,3 +7,4 @@
...
 {"ts":1641910768.499966,"uid":"CZVlyH3mlEGvfFfTJh","id.orig_h":"10.156.0.2","id.orig_p":52962,"id.resp_h":"10.156.0.3","id.resp_p":7000,"trans_depth":1,"method":"GET","uri":"/info?txtAirPlay&txtRAOP RTSP/1.","request_body_len":0,"response_body_len":0,"tags":[]}
+{"ts":1641910768.499966,"uid":"CZVlyH3mlEGvfFfTJh","id.orig_h":"10.156.0.2","id.orig_p":52962,"id.resp_h":"10.156.0.3","id.resp_p":7000,"trans_depth":1,"method":"GET","uri":"http://msedge.b.tlw.d2.delivery.mp.microsoft.com/filestreamingservice/files/48327492874837349827394-ajsdbasdbisuabdiqsnakjndaks-7634287634872368462/","request_body_len":0,"response_body_len":0,"tags":[],"host":"msedge.b.tlw.d2.delivery.mp.microsoft.com"}
```
Comment (bhapas) [inline packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log:10]: Can we add an event where `uri.domain == null` as well?

File: packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log-expected.json [hunk 2]
```diff
@@ -855,6 +855,77 @@
...
+            ],
+            "url": {
+                "domain": "msedge.b.tlw.d2.delivery.mp.microsoft.com",
+                "original": "http://msedge.b.tlw.d2.delivery.mp.microsoft.com/filestreamingservice/files/48327492874837349827394-ajsdbasdbisuabdiqsnakjndaks-7634287634872368462/",
+                "path": "/filestreamingservice/files/48327492874837349827394-ajsdbasdbisuabdiqsnakjndaks-7634287634872368462/",
+                "scheme": "http"
+            },
+            "zeek": {
+                "http": {
+                    "tags": [],
```
Comment (bhapas) [inline packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log-expected.json:924]: Do we need to have these`empty tags` or can we remove this considering a `noise` ?

File: packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log (hunk 1 above)
Comment (efd6) [reply packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log:10]: That's an existing test on line 2, unless I'm missing what you mean.

File: packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log-expected.json (hunk 2 above)
Comment (efd6) [reply packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log-expected.json:924]: That's for a separate PR if necessary.

File: packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log (hunk 1 above)
Comment (bhapas) [reply packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log:10]: Wanted to look at the test which specifically tested this line 
https://github.com/elastic/integrations/pull/6959/files/3cdfb880fcc2e39e01edf1b86fe66dfc4d0e6f38#diff-33b5627dbfbd9c74c4bbc9256e856b195a838cc26cfd40b4481214ad30ecd949R146

As it looks none of the other documents got modified in pipeline test expected log. So `rename` `skip` scenario when `ctx.zeek?.http?.host != null && ctx?.url?.domain == null` is `false` is not tested then?

File: packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log-expected.json (hunk 2 above)
Comment (bhapas) [reply packages/zeek/data_stream/http/_dev/test/pipeline/test-http.log-expected.json:924]: @andrewkroh Should we have `empty tags` a.k.a `tags: [ ]` in the output? And it is coming from[ input log](https://github.com/elastic/integrations/pull/6959/files#diff-0ae121023cbf89193f93c2f04cbe8ced7d888bca58bd2a9fa93b30729d12af5cR11) though `"tags":[]}`

**Review by bhapas — APPROVED** [review]
LGTM

**Pull request merged**